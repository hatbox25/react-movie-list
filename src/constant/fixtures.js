export const movieList = [
  {
    Title: 'Batman Begins',
    Year: '2005',
    imdbID: 'tt0372784',
    Type: 'movie',
    Poster:
      'https://m.media-amazon.com/images/M/MV5BOTY4YjI2N2MtYmFlMC00ZjcyLTg3YjEtMDQyM2ZjYzQ5YWFkXkEyXkFqcGdeQXVyMTQxNzMzNDI@._V1_SX300.jpg',
  },
  {
    Title: 'Batman Begins',
    Year: '2005',
    imdbID: 'tt0450392',
    Type: 'game',
    Poster:
      'https://m.media-amazon.com/images/M/MV5BYWRkMGNjYmQtMTJlMi00ODgyLWIyZGEtOTZkY2FlZWI3NzIwXkEyXkFqcGdeQXVyNDEyNjEzOTg@._V1_SX300.jpg',
  },
  {
    Title: 'Batman: The Journey Begins',
    Year: '2005',
    imdbID: 'tt0485593',
    Type: 'movie',
    Poster:
      'https://m.media-amazon.com/images/M/MV5BYTc0MjQ1MTctZDNlMy00NTZjLTg1MjMtN2NkMGI2ZDAwYjg3XkEyXkFqcGdeQXVyMTY4MjQ0NzU@._V1_SX300.jpg',
  },
  {
    Title: 'Batman Begins: Path to Discovery',
    Year: '2005',
    imdbID: 'tt0485689',
    Type: 'movie',
    Poster:
      'https://m.media-amazon.com/images/M/MV5BNzRlMzBjYjItMmI0YS00MjJiLThiZmUtNGFkMWRkNzAwNjQxXkEyXkFqcGdeQXVyMTY4MjQ0NzU@._V1_SX300.jpg',
  },
  {
    Title: 'Batman Begins: Behind the Mask',
    Year: '2005',
    imdbID: 'tt1074939',
    Type: 'movie',
    Poster: 'N/A',
  },
  {
    Title: 'Batman Begins: Reflections on Writing',
    Year: '2005',
    imdbID: 'tt6010256',
    Type: 'movie',
    Poster:
      'https://m.media-amazon.com/images/M/MV5BY2Y2N2EzM2YtZjdhNC00MTYwLWJiNDgtMDVlZDZjMDRlMDY1XkEyXkFqcGdeQXVyMTY4MjQ0NzU@._V1_SX300.jpg',
  },
  {
    Title: 'Batman Begins Stunts',
    Year: '2005',
    imdbID: 'tt6010270',
    Type: 'movie',
    Poster: 'N/A',
  },
  {
    Title: 'Terrible Batman Begins Remake',
    Year: '2007',
    imdbID: 'tt11723578',
    Type: 'movie',
    Poster: 'N/A',
  },
];
