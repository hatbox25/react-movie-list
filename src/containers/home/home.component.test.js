import React from 'react';
import { shallow } from 'enzyme';
import thunk from 'redux-thunk';
import toJson from 'enzyme-to-json';
import configureStore from 'redux-mock-store';

import { movieList } from '../../constant/fixtures';

import HomePageContainer from './home.container';
import HomePage from './home.component';

const mockStore = configureStore([thunk]);
const initialState = {
  movie: {
    data: movieList,
    suggestion: movieList,
    loading: false,
    hasMore: true,
  },
};
const store = mockStore(initialState);

let wrapper;
let instance;

const props = {
  suggestion: movieList,
  movies: movieList,
  loading: false,
  hasMore: false,
  getMovies: jest.fn(),
  getSuggestion: jest.fn(),
};

beforeEach(() => {
  store.dispatch = jest.fn();
  wrapper = shallow(<HomePage {...props} />);
  instance = wrapper.instance();
});

afterEach(() => {
  jest.clearAllMocks();
});

describe('HomePage Unit Test', () => {
  describe('onKeywordChange', () => {
    test('Should set correct state', () => {
      wrapper.setState({ keyword: '' });

      instance.onKeywordChange('newKeyword');

      expect(wrapper.state('keyword')).toEqual('newKeyword');
    });
  });

  describe('loadMore', () => {
    test('Should set correct state & load more movie', () => {
      wrapper.setState({ keyword: 'keyword', page: 2 });

      instance.loadMore();

      expect(wrapper.state('page')).toEqual(3);
      expect(props.getMovies).toBeCalledWith('keyword', 3);
    });
  });
});

describe('HomePage Container Snap Test', () => {
  describe('render()', () => {
    test('renders the component', () => {
      wrapper = shallow(<HomePageContainer store={store} {...props} />).dive();
      expect(toJson(wrapper)).toMatchSnapshot();
    });
  });
});
