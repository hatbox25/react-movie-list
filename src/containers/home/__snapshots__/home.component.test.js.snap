// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomePage Container Snap Test render() renders the component 1`] = `
<Home
  getMovies={[Function]}
  getSuggestion={[Function]}
  hasMore={true}
  loading={false}
  movies={
    Array [
      Object {
        "Poster": "https://m.media-amazon.com/images/M/MV5BOTY4YjI2N2MtYmFlMC00ZjcyLTg3YjEtMDQyM2ZjYzQ5YWFkXkEyXkFqcGdeQXVyMTQxNzMzNDI@._V1_SX300.jpg",
        "Title": "Batman Begins",
        "Type": "movie",
        "Year": "2005",
        "imdbID": "tt0372784",
      },
      Object {
        "Poster": "https://m.media-amazon.com/images/M/MV5BYWRkMGNjYmQtMTJlMi00ODgyLWIyZGEtOTZkY2FlZWI3NzIwXkEyXkFqcGdeQXVyNDEyNjEzOTg@._V1_SX300.jpg",
        "Title": "Batman Begins",
        "Type": "game",
        "Year": "2005",
        "imdbID": "tt0450392",
      },
      Object {
        "Poster": "https://m.media-amazon.com/images/M/MV5BYTc0MjQ1MTctZDNlMy00NTZjLTg1MjMtN2NkMGI2ZDAwYjg3XkEyXkFqcGdeQXVyMTY4MjQ0NzU@._V1_SX300.jpg",
        "Title": "Batman: The Journey Begins",
        "Type": "movie",
        "Year": "2005",
        "imdbID": "tt0485593",
      },
      Object {
        "Poster": "https://m.media-amazon.com/images/M/MV5BNzRlMzBjYjItMmI0YS00MjJiLThiZmUtNGFkMWRkNzAwNjQxXkEyXkFqcGdeQXVyMTY4MjQ0NzU@._V1_SX300.jpg",
        "Title": "Batman Begins: Path to Discovery",
        "Type": "movie",
        "Year": "2005",
        "imdbID": "tt0485689",
      },
      Object {
        "Poster": "N/A",
        "Title": "Batman Begins: Behind the Mask",
        "Type": "movie",
        "Year": "2005",
        "imdbID": "tt1074939",
      },
      Object {
        "Poster": "https://m.media-amazon.com/images/M/MV5BY2Y2N2EzM2YtZjdhNC00MTYwLWJiNDgtMDVlZDZjMDRlMDY1XkEyXkFqcGdeQXVyMTY4MjQ0NzU@._V1_SX300.jpg",
        "Title": "Batman Begins: Reflections on Writing",
        "Type": "movie",
        "Year": "2005",
        "imdbID": "tt6010256",
      },
      Object {
        "Poster": "N/A",
        "Title": "Batman Begins Stunts",
        "Type": "movie",
        "Year": "2005",
        "imdbID": "tt6010270",
      },
      Object {
        "Poster": "N/A",
        "Title": "Terrible Batman Begins Remake",
        "Type": "movie",
        "Year": "2007",
        "imdbID": "tt11723578",
      },
    ]
  }
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [MockFunction],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
  suggestion={
    Array [
      Object {
        "Poster": "https://m.media-amazon.com/images/M/MV5BOTY4YjI2N2MtYmFlMC00ZjcyLTg3YjEtMDQyM2ZjYzQ5YWFkXkEyXkFqcGdeQXVyMTQxNzMzNDI@._V1_SX300.jpg",
        "Title": "Batman Begins",
        "Type": "movie",
        "Year": "2005",
        "imdbID": "tt0372784",
      },
      Object {
        "Poster": "https://m.media-amazon.com/images/M/MV5BYWRkMGNjYmQtMTJlMi00ODgyLWIyZGEtOTZkY2FlZWI3NzIwXkEyXkFqcGdeQXVyNDEyNjEzOTg@._V1_SX300.jpg",
        "Title": "Batman Begins",
        "Type": "game",
        "Year": "2005",
        "imdbID": "tt0450392",
      },
      Object {
        "Poster": "https://m.media-amazon.com/images/M/MV5BYTc0MjQ1MTctZDNlMy00NTZjLTg1MjMtN2NkMGI2ZDAwYjg3XkEyXkFqcGdeQXVyMTY4MjQ0NzU@._V1_SX300.jpg",
        "Title": "Batman: The Journey Begins",
        "Type": "movie",
        "Year": "2005",
        "imdbID": "tt0485593",
      },
      Object {
        "Poster": "https://m.media-amazon.com/images/M/MV5BNzRlMzBjYjItMmI0YS00MjJiLThiZmUtNGFkMWRkNzAwNjQxXkEyXkFqcGdeQXVyMTY4MjQ0NzU@._V1_SX300.jpg",
        "Title": "Batman Begins: Path to Discovery",
        "Type": "movie",
        "Year": "2005",
        "imdbID": "tt0485689",
      },
      Object {
        "Poster": "N/A",
        "Title": "Batman Begins: Behind the Mask",
        "Type": "movie",
        "Year": "2005",
        "imdbID": "tt1074939",
      },
      Object {
        "Poster": "https://m.media-amazon.com/images/M/MV5BY2Y2N2EzM2YtZjdhNC00MTYwLWJiNDgtMDVlZDZjMDRlMDY1XkEyXkFqcGdeQXVyMTY4MjQ0NzU@._V1_SX300.jpg",
        "Title": "Batman Begins: Reflections on Writing",
        "Type": "movie",
        "Year": "2005",
        "imdbID": "tt6010256",
      },
      Object {
        "Poster": "N/A",
        "Title": "Batman Begins Stunts",
        "Type": "movie",
        "Year": "2005",
        "imdbID": "tt6010270",
      },
      Object {
        "Poster": "N/A",
        "Title": "Terrible Batman Begins Remake",
        "Type": "movie",
        "Year": "2007",
        "imdbID": "tt11723578",
      },
    ]
  }
/>
`;
